service/discovery/discovery.go

```diff
@@ -434,6 +456,16 @@ func (svc *Service) Query(_ context.Context, req *discovery.ListResourcesRequest
 		}
 	}
 
+	// We need to further restrict our query according to the cloud service we are allowed to "see.".
+	//
+	// TODO(oxisto): This is suboptimal, since we are now calling AllowedCloudServices twice. Once here
```

Comment:
I think you cannot get rid of the two callings, since you change the CheckAccess method. If  CheckAccess() also returns the "all"-bool you could simplify the checks here.

Reply:
Yeah that would probably solve it, but then I would want to overhaul the `CheckAccess` in all calls and let's do that separately then.